framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4631,6 +4613,442 @@ private void makeGroundTargetType(
     addDefaultAnnotations(functionalType);
   }
 
+  /**
+   * Return true if {@code type} should be captured.
+   *
+   * <ul>
+   *   <li>{@code type} and {@code typeMirror} must both be declared types.
+   *   <li>{@code type} must not be a raw type, must not have an uninferred type argument, and must
+   *       have a wildcard as a type argument.
+   *   <li>{@code typeMirror} must have a captured type as a type argument.
```

Comment:
Please state the relationship between `type` and `typeMirror`.  From examining calls, I think that typeMirror must correspond to type and must be a captured type (that is, those are preconditions of the method).